src/azure-cli/azure/cli/command_modules/appservice/access_restrictions.py

```diff
@@ -0,0 +1,163 @@
+    if subnet or vnet_name:
+        subnet_id = _validate_subnet(cmd.cli_ctx, subnet, vnet_name, resource_group_name)
+        if not ignore_missing_vnet_service_endpoint:
+            _ensure_subnet_service_endpoint(cmd.cli_ctx, subnet_id)
+
+        for rule in list(access_rules):
+            if rule.action == action and rule.vnet_subnet_resource_id == subnet_id:
+                rule_instance = rule
+                break
+
+        if rule_instance:
+            rule_instance.name = rule_name
+            rule_instance.priority = priority
+            rule_instance.description = description if description else rule_instance.description
+        else:
+            rule_instance = IpSecurityRestriction(
+                name=rule_name, vnet_subnet_resource_id=subnet_id,
+                priority=priority, action=action, tag='Default', description=description)
+            access_rules.append(rule_instance)
+
+    if ip_address:
+        for rule in list(access_rules):
+            if rule.action == action and rule.ip_address == ip_address:
+                rule_instance = rule
+                break
+
+        if rule_instance:
+            rule_instance.name = rule_name
+            rule_instance.priority = priority
+            rule_instance.description = description if description else rule_instance.description
+        else:
+            rule_instance = IpSecurityRestriction(
+                name=rule_name, ip_address=ip_address,
+                priority=priority, action=action, tag='Default', description=description)
+            access_rules.append(rule_instance)
+
+    result = _generic_site_operation(
+        cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
+    return result.scm_ip_security_restrictions if scm_site else result.ip_security_restrictions
+
+
+def remove_webapp_access_restriction(cmd, resource_group_name, name, rule_name, scm_site=False, slot=None):
+    configs = get_site_configs(cmd, resource_group_name, name, slot)
+    rule_instance = None
+    # get rules list
+    access_rules = configs.scm_ip_security_restrictions if scm_site else configs.ip_security_restrictions
+
+    for rule in list(access_rules):
+        if rule.name == rule_name:
+            rule_instance = rule
+            break
+
+    if rule_instance is not None:
+        access_rules.remove(rule_instance)
+
+    result = _generic_site_operation(
+        cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
+    return result.scm_ip_security_restrictions if scm_site else result.ip_security_restrictions
+
+
+def set_webapp_access_restriction(cmd, resource_group_name, name, use_same_restrictions_for_scm_site, slot=None):
+    configs = get_site_configs(cmd, resource_group_name, name, slot)
+    if use_same_restrictions_for_scm_site:
+        setattr(configs, 'scm_ip_security_restrictions_use_main', True)
+    else:
+        setattr(configs, 'scm_ip_security_restrictions_use_main', False)
```

Comment:
maybe just `setattr(configs, 'scm_ip_security_restrictions_use_main', bool(use_same_restrictions_for_scm_site))`?

Reply:
Makes sense - and easier to read.